### PR TITLE
Explicitly state what the conversation.wait() method returns

### PR DESCRIPTION
**Pull request by Aquathing**

### CODE REVIEW

**Review by KnorpelSenf — CHANGES_REQUESTED** [review]
I'd like to see something added along the lines of, __that means you're always dealing with as many context objects at the same time as updates are received during the conversation__ or so.

Also, perhaps we could add another wait call to the above code snippet to further illustrate this point?

**Review by KnorpelSenf — APPROVED** [review]
Well done, thank you so much!

**Comment by KnorpelSenf** [comment]
Are you able to fix CI?

File: site/docs/plugins/conversations.md [hunk 1]
```diff
@@ -571,7 +571,63 @@ async function waitForMe(conversation, ctx) {
...
+The wait method always yields a new [context](https://grammy.dev/guide/context.html) object representing the received update.
+That means you're always dealing with as many context objects as updates received during the conversation.
 
+<CodeGroup>
+<CodeGroupItem title="TypeScript" active>
+
+```ts
+const TEAM_REVIEW_CHAT = -10018121912;
+async function askUser(conversation: MyConversation, ctx: MyContext) {
+  // Ask the user for their home address
```
Comment (roj1512) [inline site/docs/plugins/conversations.md:584]:
```suggestion
  // Ask the user for their home address.
```

File: site/docs/plugins/conversations.md [hunk 2]
```diff
@@ -571,7 +571,63 @@ async function waitForMe(conversation, ctx) {
...
+  await ctx.reply("Could you state your home address?");
+  
+  // Wait for the user to send their address:
+  const userHomeAddressContext = await conversation.wait();
+  
+  // Ask the user their nationality
```
Comment (roj1512) [inline site/docs/plugins/conversations.md:590]:
```suggestion
  // Ask the user about their nationality.
```

File: site/docs/plugins/conversations.md [hunk 3]
```diff
@@ -571,7 +571,63 @@ async function waitForMe(conversation, ctx) {
...
+  await ctx.reply("Could you also please state your nationality?");
+  
+  // Wait for the user to state their nationality
+  const userNationalityContext = await conversation.wait();
+  await ctx.reply("That was the final step. Now that I've received all relevant information I'll forward them to our team for review. Thank you!");
+  
+  // Now, we copy both messages to our review chat
```
Comment (roj1512) [inline site/docs/plugins/conversations.md:597]:
```suggestion
  // Now, we copy the responses to another chat for review.
```

File: site/docs/plugins/conversations.md [hunk 4]
```diff
@@ -571,7 +571,63 @@ async function waitForMe(conversation, ctx) {
...
+}
+```
+
+</CodeGroupItem>
+ <CodeGroupItem title="JavaScript">
+
+```js
+const TEAM_REVIEW_CHAT = -10018121912;
+async function askUser(conversation, ctx) {
+  // Ask the user for their home address
```
Comment (roj1512) [inline site/docs/plugins/conversations.md:609]:
```suggestion
  // Ask about the user's home address.
```

File: site/docs/plugins/conversations.md [hunk 5]
```diff
@@ -571,7 +571,63 @@ async function waitForMe(conversation, ctx) {
...
+  await ctx.reply("Could you state your home address?");
+  
+  // Wait for the user to send their address:
+  const userHomeAddressContext = await conversation.wait();
+  
+  // Ask the user their nationality
```
Comment (roj1512) [inline site/docs/plugins/conversations.md:615]:
```suggestion
  // Ask the user about their nationality.
```

File: site/docs/plugins/conversations.md [hunk 6]
```diff
@@ -571,7 +571,63 @@ async function waitForMe(conversation, ctx) {
...
+  await ctx.reply("Could you also please state your nationality?");
+  
+  // Wait for the user to state their nationality
```
Comment (roj1512) [inline site/docs/plugins/conversations.md:618]:
```suggestion
  // Wait for the user to state their nationality.
```

File: site/docs/plugins/conversations.md [hunk 7]
```diff
@@ -571,7 +571,63 @@ async function waitForMe(conversation, ctx) {
...
+  const userNationalityContext = await conversation.wait();
+  await ctx.reply("That was the final step. Now that I've received all relevant information I'll forward them to our team for review. Thank you!");
+  
+  // Now, we copy both messages to our review chat
```
Comment (roj1512) [inline site/docs/plugins/conversations.md:622]:
```suggestion
  // Now, we copy the responses to another chat for review.
```

File: site/docs/plugins/conversations.md [hunk 8]
```diff
@@ -571,7 +571,63 @@ async function waitForMe(conversation, ctx) {
 
 An update can mean that a text message was sent, or a button was pressed, or something was edited, or virtually any other action was performed by the user.
 Check out the full list in the Telegram docs [here](https://core.telegram.org/bots/api#update).
+   
+The wait method always yields a new [context](https://grammy.dev/guide/context.html) object representing the received update.
+That means you're always dealing with as many context objects as updates received during the conversation.
```
Comment (roj1512) [inline site/docs/plugins/conversations.md:576]: As updates received?

File: site/docs/plugins/conversations.md (hunk 2 above)
Comment (Aquathing) [reply site/docs/plugins/conversations.md:590]: Why "about"?

File: site/docs/plugins/conversations.md (hunk 8 above)
Comment (Aquathing) [reply site/docs/plugins/conversations.md:576]: It didn't sound wrong. I'll fix it.

File: site/docs/plugins/conversations.md (hunk 2 above)
Comment (KnorpelSenf) [reply site/docs/plugins/conversations.md:590]: You can also use “for,” but a preposition is required here.

File: site/docs/plugins/conversations.md (hunk 5 above)
Comment (Aquathing) [reply site/docs/plugins/conversations.md:615]: I don't see why "about"? We just need their nationality

File: site/docs/plugins/conversations.md (hunk 8 above)
Comment (KnorpelSenf) [reply site/docs/plugins/conversations.md:576]:
```suggestion
That means you're always dealing with as many context objects as there are updates received during the conversation.
```

**Comment by KnorpelSenf** [comment]
Note that from the Files changed tab, you can add several suggestions to a batch and commit them in bulk in a single commit.

File: site/docs/plugins/conversations.md (hunk 5 above)
Comment (KnorpelSenf) [reply site/docs/plugins/conversations.md:615]: Same as above in https://github.com/grammyjs/website/pull/578#discussion_r1064015754

File: site/docs/plugins/conversations.md (hunk 1 above)
Comment (KnorpelSenf) [reply site/docs/plugins/conversations.md]: Wouldn't _for_ be more appropriate than _about_ here?

File: site/docs/plugins/conversations.md (hunk 2 above)
Comment (Aquathing) [reply site/docs/plugins/conversations.md:590]: So would "Ask the user for their nationality" be good?

Comment (KnorpelSenf) [reply site/docs/plugins/conversations.md:590]: IMO yes.

File: site/docs/plugins/conversations.md [hunk 9]
```diff
@@ -571,7 +571,64 @@ async function waitForMe(conversation, ctx) {
 
 An update can mean that a text message was sent, or a button was pressed, or something was edited, or virtually any other action was performed by the user.
 Check out the full list in the Telegram docs [here](https://core.telegram.org/bots/api#update).
+   
+The wait method always yields a new [context](https://grammy.dev/guide/context.html) object representing the received update.
```
Comment (CikiMomogi) [inline site/docs/plugins/conversations.md:575]:
```suggestion
The wait method always yields a new [context](../guide/context.md) object representing the received update.
```

File: site/docs/plugins/conversations.md [hunk 10]
```diff
@@ -571,7 +571,64 @@ async function waitForMe(conversation, ctx) {
...
+  // Ask the user for their home address.
+  await ctx.reply("Could you state your home address?");
+  
+  // Wait for the user to send their address:
+  const userHomeAddressContext = await conversation.wait();
+  
+  // Ask the user for their nationality.
+  await ctx.reply("Could you also please state your nationality?");
+  
+  // Wait for the user to state their nationality:
```
Comment (roj1512) [inline site/docs/plugins/conversations.md:593]: Why a colon here?

**Review by CikiMomogi — CHANGES_REQUESTED** [review]

File: site/docs/plugins/conversations.md [hunk 11]
```diff
@@ -571,7 +571,64 @@ async function waitForMe(conversation, ctx) {
...
+<CodeGroupItem title="TypeScript" active>
+
+```ts
+const TEAM_REVIEW_CHAT = -10018121912;
+async function askUser(conversation: MyConversation, ctx: MyContext) {
+  // Ask the user for their home address.
+  await ctx.reply("Could you state your home address?");
+  
+  // Wait for the user to send their address:
+  const userHomeAddressContext = await conversation.wait();
```
Comment (Loskir) [inline site/docs/plugins/conversations.md:588]: Why context and not ctx?

File: site/docs/plugins/conversations.md (hunk 5 above)
Comment (roj1512) [reply site/docs/plugins/conversations.md]: We'll go with the "for" for now.

File: site/docs/plugins/conversations.md (hunk 2 above)
Comment (roj1512) [reply site/docs/plugins/conversations.md]: https://github.com/grammyjs/website/pull/578#discussion_r1064026055

File: site/docs/plugins/conversations.md (hunk 1 above)
Comment (roj1512) [reply site/docs/plugins/conversations.md]: Let's go with that.

File: site/docs/plugins/conversations.md (hunk 4 above)
Comment (roj1512) [reply site/docs/plugins/conversations.md]: https://github.com/grammyjs/website/pull/578#discussion_r1064026091

File: site/docs/plugins/conversations.md (hunk 10 above)
Comment (Aquathing) [reply site/docs/plugins/conversations.md:593]: It uses a colon there in the previous example, so I decided to use it here as well

File: site/docs/plugins/conversations.md (hunk 11 above)
Comment (Aquathing) [reply site/docs/plugins/conversations.md:588]: It's just easier to read, it's an example after all

File: site/docs/plugins/conversations.md [hunk 12]
```diff
@@ -571,7 +571,64 @@ async function waitForMe(conversation, ctx) {
...
+  
+  // Ask the user for their nationality.
+  await ctx.reply("Could you also please state your nationality?");
+  
+  // Wait for the user to state their nationality:
+  const userNationalityContext = await conversation.wait();
+  
+  await ctx.reply("That was the final step. Now that I've received all relevant information I'll forward them to our team for review. Thank you!");
+  
+  // Now, we copy the responses to another chat for review.
```
Comment (roj1512) [inline site/docs/plugins/conversations.md:598]:
```suggestion
  // We now copy the responses to another chat for review.
```

File: site/docs/plugins/conversations.md [hunk 13]
```diff
@@ -571,7 +571,64 @@ async function waitForMe(conversation, ctx) {
...
+  const userHomeAddressContext = await conversation.wait();
+  
+  // Ask the user for their nationality.
+  await ctx.reply("Could you also please state your nationality?");
+  
+  // Wait for the user to state their nationality:
+  const userNationalityContext = await conversation.wait();
+  await ctx.reply("That was the final step. Now that I've received all relevant information I'll forward them to our team for review. Thank you!");
+  
+  // Now, we copy the responses to another chat for review.
```
Comment (roj1512) [inline site/docs/plugins/conversations.md:623]:
```suggestion
  // We now copy the responses to another chat for review.
```

File: site/docs/plugins/conversations.md (hunk 10 above)
Comment (roj1512) [reply site/docs/plugins/conversations.md:593]: Fair.

**Review by roj1512 — APPROVED** [review]
LGTM. Thank you so much!

**Comment by Aquathing** [comment]
Thank you all for the reviews!

**Comment by roj1512** [comment]
And you, for the PQ!

File: site/docs/plugins/conversations.md [hunk 14]
```diff
@@ -572,6 +572,68 @@ async function waitForMe(conversation, ctx) {
...
 Check out the full list in the Telegram docs [here](https://core.telegram.org/bots/api#update).
 
+The `wait` method always yields a new [context object](../guide/context.md) representing the received update.
+That means you're always dealing with as many context objects as there are updates received during the conversation.
+
+<CodeGroup>
+<CodeGroupItem title="TypeScript" active>
+
+```ts
+const TEAM_REVIEW_CHAT = -10018121912;
```
Comment (CikiMomogi) [inline site/docs/plugins/conversations.md:582]: Can you tell us who this ID belongs to? I understand that chat IDs are not private information and can be shared publicly, but would it be better for us to replace it with an obvious random chat ID, like -9991234567890, or even better, promote our Telegram channel, like -1001402419718, which is the ID for the grammY News?

Comment (roj1512) [reply site/docs/plugins/conversations.md:582]: Good idea!

**Comment by roj1512** [comment]
Will get translated to Spanish in https://github.com/grammyjs/website/pull/582.

**Comment by CikiMomogi** [comment]
Let me sync the changes to ID

**Comment by roj1512** [comment]
What changes?

**Review by CikiMomogi — APPROVED** [review]

**Pull request merged**